cmake/extensions.cmake

```diff
@@ -848,8 +848,22 @@ function(import_kconfig prefix kconfig_fragment)
       set(CONF_VARIABLE_VALUE ${CMAKE_MATCH_1})
     endif()
 
-    set("${CONF_VARIABLE_NAME}" "${CONF_VARIABLE_VALUE}" PARENT_SCOPE)
+    if (${property})
+      set_property(${ARGN} "${CONF_VARIABLE_NAME}" "${CONF_VARIABLE_VALUE}")
+    else()
+      set("${CONF_VARIABLE_NAME}" "${CONF_VARIABLE_VALUE}" PARENT_SCOPE)
+    endif()
```

Comment:
This use-case of needing to set properties in addition to variables shows that this function is doing too many things.

Let us instead have a function that does one thing, determines what the key-value set from a kconfig_fragment is. And then have either users use this key-value set directly or add wrapper-functions for each use-case (depending on how common it is). E.g.

```
# The last two arguments are output parameters
function(read_kconfig prefix kconfig_fragment keys values)

# Uses read_kconfig to populate the namespace (usually with CONF_ variables)
function(import_kconfig prefix kconfig_fragment)
```

Reply:
I really like this idea, but I ended up with something slightly different. When I implemented your suggestion, there was quite a few lines of boilerplate to iterate over the lists in parallell. Instead, I modified the existing import_kconfig() so it can return a list of the variables it set. This makes the change much smaller.